### PR TITLE
ShowHiddenChannels: New screen for showing hidden channels

### PR DESCRIPTION
**Pull request by Nuckyz**
image
image

### CODE REVIEW

File: src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+}));
+
+const ForumLayoutTypesToNames = proxyLazy(() => ({
+    [ForumLayoutTypes.Default]: "Not set",
+    [ForumLayoutTypes.LIST]: "List view",
+    [ForumLayoutTypes.GRID]: "Gallery view"
+}));
+
+// Icon from the modal when clicking a message link you don't have access to view
+const HiddenChannelLogo = "/assets/433e3ec4319a9d11b0cbe39342614982.svg";
```
Comment (Vendicated) [inline src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx:56]: is hardcoding this safe?

File: src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+function HiddenChannelLockScreen(channel: any) {
```
Comment (Vendicated) [inline src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx:58]: why not use the channel type from discord-types?

File: src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                            )}
+                        </Tooltip>
+                    }
+                </div>
+
+                <Text variant="text-lg/normal">You can not see the {channel.isForumChannel() ? "posts" : "messages"} of this channel. {channel.isForumChannel() && channel.topic?.length > 0 && "However you may see its guidelines:"}</Text >
+
+                {channel.isForumChannel() && channel.topic?.length > 0 && <div className="shc-lock-screen-topic-container">
+                    {Parser.parseTopic(channel.topic, false, { channelId: channel.id })}
+                </div>}
```
Comment (Vendicated) [inline src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx:90]: why do you only show the topic of forum channels and not regular channels?

File: src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx (hunk 1 above)
Comment (Nuckyz) [reply src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx:56]: I would have to the match the url from the module which isn't exported and is lazy loaded

File: src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx (hunk 2 above)
Comment (Nuckyz) [reply src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx:58]: misses some types related to forums

File: src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx (hunk 3 above)
Comment (Nuckyz) [reply src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx:90]: Regular channels already show the topic on the header, I'm not removing that component. However forum channels do not show it there

Comment (Vendicated) [reply src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx:90]: I see

File: src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx (hunk 2 above)
Comment (Vendicated) [reply src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx:58]: could just `Channel & { myChannelExtension?(): void }`

Comment (Nuckyz) [reply src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx:58]: sure I will do it later

Comment (Nuckyz) [reply src/plugins/showHiddenChannels/components/HiddenChannelLockScreen.tsx]: Done, also fixed a typo I found in the process lol

**Comment by Nuckyz** [comment]
Aight this is for sure ready now, there is nothing else to add

File: src/plugins/showHiddenChannels/util.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    years: { value: 31536000, unit: "year" },
+    months: { value: 2592000, unit: "month" },
+    days: { value: 86400, unit: "day" },
+    hours: { value: 3600, unit: "hour" },
+    minutes: { value: 60, unit: "minute" },
+    seconds: { value: 1, unit: "second" },
+    milliseconds: { value: 0.001, unit: "millisecond" }
+};
+
+export function humanizeTime(time: number, unit: Units): string {
```
Comment (Vendicated) [inline src/plugins/showHiddenChannels/util.ts:36]: i don't like this, why not use moment for this?

Comment (Vendicated) [reply src/plugins/showHiddenChannels/util.ts:36]: or Intl https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat

Comment (Nuckyz) [reply src/plugins/showHiddenChannels/util.ts:36]: moment seems to follow Discord thresholds and refuses to change its settings (it turns 30 seconds into a few seconds)

And I couldn't find a way to do it with Intl

**Pull request merged**